src/Core/src/Platform/iOS/WindowExtensions.cs

```diff
@@ -15,23 +16,49 @@ internal static void UpdateTitle(this UIWindow platformWindow, IWindow window)
 			// If you set it to an empty string the title reverts back to the 
 			// default app title.
 			if (OperatingSystem.IsIOSVersionAtLeast(13) && platformWindow.WindowScene is not null)
+			{
 				platformWindow.WindowScene.Title = window.Title ?? String.Empty;
+			}
 		}
 
 		internal static void UpdateX(this UIWindow platformWindow, IWindow window) =>
-			platformWindow.UpdateUnsupportedCoordinate(window);
+			platformWindow.UpdateCoordinates(window);
 
 		internal static void UpdateY(this UIWindow platformWindow, IWindow window) =>
-			platformWindow.UpdateUnsupportedCoordinate(window);
+			platformWindow.UpdateCoordinates(window);
 
 		internal static void UpdateWidth(this UIWindow platformWindow, IWindow window) =>
-			platformWindow.UpdateUnsupportedCoordinate(window);
+			platformWindow.UpdateCoordinates(window);
 
 		internal static void UpdateHeight(this UIWindow platformWindow, IWindow window) =>
-			platformWindow.UpdateUnsupportedCoordinate(window);
+			platformWindow.UpdateCoordinates(window);
+
+		internal static void UpdateCoordinates(this UIWindow platformWindow, IWindow window)
+		{
+			var rectangle = platformWindow.Bounds.ToRectangle();
 
-		internal static void UpdateUnsupportedCoordinate(this UIWindow platformWindow, IWindow window) =>
-			window.FrameChanged(platformWindow.Bounds.ToRectangle());
+			if (OperatingSystem.IsIOSVersionAtLeast(16)) 
+			{
+				var windowScene = platformWindow.WindowScene;
+
+				if (windowScene is not null)
+				{
+					var preferences = new UIWindowSceneGeometryPreferencesMac
+					{
+						SystemFrame = new CGRect(rectangle.X, rectangle.Y, rectangle.Width, rectangle.Height)
+					};
+
+					windowScene.RequestGeometryUpdate(preferences, (error) => {
+						System.Diagnostics.Debug.WriteLine("" + error);
+					});
+				}
+			}
+			else 
+			{
+				// TODO: When should this be called exactly?
+				window.FrameChanged(rectangle);
```

Comment:
This should be called to update the system if the window is not changing. For example, this till needs to be called if the window scene is null.

The just "fakes" the event from the OS after the window changes. We need to make sure the `RequestGeometryUpdate` triggers the right events and fires all the Frame and size changed events. If you set width to say 300, do the size events fire? Is there notification from the OS back into MAUI?